src/Symfony/Component/Validator/Constraints/DivisibleByValidator.php

```diff
@@ -23,6 +23,10 @@ class DivisibleByValidator extends AbstractComparisonValidator
      */
     protected function compareValues($value1, $value2)
     {
+        if (!is_numeric($value1) || !is_numeric($value2)) {
+            return false;
```

Comment:
shouldnt we throw `UnexpectedValueException` :thinking:

Reply:
We might as well, but soft fail is fine to me.

Reply:
yes, but `UnexpectedValueException` leads to a better violation

https://github.com/symfony/symfony/blob/03bb2da078aea1cd4e548b930f68a2c01083963e/src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php#L730-L736

Reply:
Well, passing an object shouldn't be expected here, so I think what Roland suggested is correct. It isn't the same, return false because it's not divisible than returning false because we cannot determine it. It seems that `UnexpectedValueException` fits better in this situation.

Reply:
Alright 😃 In my mind it was like: does my object is divisible by 2 -> no. But I didn't know about this special exception and its treatment so very good suggestion!